sdk/storage/Azure.Storage.DataMovement/src/StorageResourceBase.cs

```diff
@@ -36,5 +46,23 @@ public abstract class StorageResourceBase
         /// Defines whether the storage resource is a container.
         /// </summary>
         public abstract bool IsContainer { get; }
+
+        // TODO: add back in when AzureSasCredential supports generating SAS's
+        // <summary>
+        // Internal constructor to accept the authorization Scheme
+        // </summary>
+        //protected void SetAuthorizationScheme(AzureSasCredential sasCredential)
```

Comment:
We really shouldn't be putting SAS in the base package since that is an Azure Storage specific concept. `HttpAuthorization ` is a little more standard so I could see the argument for that one I guess.